src/main/java/bio/terra/service/filedata/DrsMetricsService.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.service.filedata;
+
+import bio.terra.app.controller.exception.TooManyRequestsException;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DrsMetricsService {
```

Comment:
This is a nice approach.  Probably not needed but just wondering if there are any other cases with call metrics like this that would benefit from a similar metrics service and it this should be made generic?

Reply:
This is a great question, and one I considered… for the time being, I chose to not over-generalize.  If there are other use cases outside of DRS code, one approach may be a base `MetricsService` from which usage-specific services extend, to share common code, naming conventions, etc.  But I could see one single `MetricsService` getting a bit unwieldy.

Reply:
Oh totally.  Perhaps extensions of an abstract class at some point. Again, more just food for thought